synapse/tools/migrate_010.py

```diff
@@ -0,0 +1,715 @@
+        'inet:udp6': convert_inet_xxp_primary,
+        'inet:tcp4': convert_inet_xxp_primary,
+        'inet:tcp6': convert_inet_xxp_primary,
+        'ps:image': convert_ps_image,
+        'file:base': check_file_base_primary,
+    }
+
+    def handle_seen(self, propname, propval, newprops):
+        '''
+        :seen:min, :seen:max -> .seen = (min, max)
+        '''
+        seenmin, seenmax = newprops.get('.seen', (None, None))
+        if propname.endswith(':min'):
+            seenmin = propval
+        if propname.endswith(':max'):
+            seenmax = propval
+        if seenmax is None:
+            seenmax = seenmin + 1
+        if seenmin is None:
+            seenmin = seenmax - 1
+        return '.seen', (seenmin, seenmax)
+
+    def ip_with_port_to_server(self, formname, propname, typename, val, props):
+        port = props.get(formname + ':port')
+        if propname[-1] == '4':
+            if port is not None:
+                val = (val << 16) + port
+            addrport = self.core.getTypeRepr('inet:srv4', val)
+        else:
+            if port is None:
+                addrport = '[' + val + ']'
+            else:
+                addrport = '[%s]:%s' % (val, port)
+        return propname, '%s://%s' % (formname[-3:], addrport)
+
+    subprop_special = {
+        'inet:exec:url:ipv4': ipv4_to_client,
+        'inet:exec:url:ipv6': ipv6_to_client,
+        'inet:web:logon:ipv4': ipv4_to_client,
+        'inet:web:logon:ipv6': ipv6_to_client,
+        'inet:web:action:ipv4': ipv4_to_client,
+        'inet:web:action:ipv6': ipv6_to_client,
+        'inet:web:acct:signup:ipv4': ipv4_to_client,
+        'inet:dns:look:ipv4': ipv4_to_client,
+        'inet:dns:look:ipv6': ipv6_to_client,
+        'it:exec:bind:tcp:ipv4': ip_with_port_to_server,
+        'it:exec:bind:tcp:ipv6': ip_with_port_to_server,
+        'it:exec:bind:udp:ipv4': ip_with_port_to_server,
+        'it:exec:bind:udp:ipv6': ip_with_port_to_server,
+    }
+
+def main(argv, outp=None):  # pragma: no cover
+    p = argparse.ArgumentParser(description='''Command line tool to export a Synapse Cortex v. 0.0.5 to a mpk file
+ for importation into a 0.1.0 cortex''')
+    p.add_argument('cortex', help='telepath URL for a cortex to be dumped')
+    p.add_argument('outfile', help='file to dump to')
+    p.add_argument('--verbose', '-v', action='count', help='Verbose output')
+    p.add_argument('--stage-1', help='Start at stage 2 with stage 1 file')
+    opts = p.parse_args(argv)
+
+    if opts.verbose is not None:
+        if opts.verbose > 1:
+            logger.setLevel(logging.DEBUG)
+        elif opts.verbose > 0:
+            logger.setLevel(logging.INFO)
+
+    fh = logging.FileHandler(opts.outfile + '.log')
+    fh.setLevel(logging.DEBUG)
+    logger.addHandler(fh)
+
+    fh = open(opts.outfile, 'wb')
+    rejects_fh = open(opts.outfile + '.rejects', 'wb')
+    with s_cortex.openurl(opts.cortex, conf={'caching': True, 'cache:maxsize': 1000000}) as core:
```

Comment:
if someone has a custom module, how would they load them?

Reply:
visi said not to worry about that.

Reply:
I think we do have to handle this

Reply:
done